GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+public final class Preconditions {
```

Comment:
Actually in Google Guava, the Preconditions class makes its constructor private which serves same purpose as your point 3.

I am thinking that if we should keep this class simple and not used that frequently so that I made it final. I think abstract is also an option. I will update the patch based on your comments.